### PR TITLE
Collapsible challenge submission table UI

### PR DESCRIPTION
**Pull request by sanketbansal**
Modified submission table into a collapsible one.

#### Changes proposed in this pull request:

- Modified challenge submission table into a collapsible one.
- Fields like status, stderr_file, stdout_file, submission_result_file are displayed outside collapsible
rest data remains inside collapsible.

- Link to live demo: http://pr-187-evalai.surge.sh 

Screenshot from 2019-07-22 17-27-45

Screenshot from 2019-07-22 16-32-03

### CODE REVIEW

**Comment by lunayach** [comment]
@sanketbansal, can you pass it's build?

**Comment by sanketbansal** [comment]
> @sanketbansal, can you pass it's build?

Looking for it

**Comment by lunayach** [comment]
@sanketbansal resolve the conflicts.

**Comment by sanketbansal** [comment]
> @sanketbansal resolve the conflicts.

@lunayach  done please review

File: src/app/app.module.ts [hunk 1]
```diff
@@ -80,6 +81,8 @@ import {
 } from './components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component';
 import { SideBarComponent } from './components/utility/side-bar/side-bar.component';
 
+import {MatTableModule} from '@angular/material/table';
```
Comment (lunayach) [inline src/app/app.module.ts]:
```suggestion
import { MatTableModule } from '@angular/material/table';
```
Uniform spacing. Also for the line which is below.

File: src/app/components/challenge/challenge.component.spec.ts [hunk 2]
```diff
@@ -24,7 +24,8 @@ import { InputComponent } from '../../components/utility/input/input.component';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
 import { Router, ActivatedRoute } from '@angular/router';
 import { Observable, of } from 'rxjs';
-import {FormsModule} from '@angular/forms';
+import {MatTableModule} from '@angular/material';
```
Comment (lunayach) [inline src/app/components/challenge/challenge.component.spec.ts]:
```suggestion
import { MatTableModule } from '@angular/material';
```

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts [hunk 3]
```diff
@@ -9,13 +9,22 @@ import { Router, ActivatedRoute } from '@angular/router';
 import { SelectphaseComponent } from '../../utility/selectphase/selectphase.component';
 import { InputComponent } from '../../utility/input/input.component';
 
+import {Observable} from 'rxjs';
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts]: Here as well. Recommend you to change your editor's settings.

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts [hunk 4]
```diff
@@ -134,6 +143,10 @@ export class ChallengesubmissionsComponent implements OnInit, AfterViewInit {
    */
   apiCall: any;
 
+  columnsToDisplay = ['status', 'stderr_file', 'stdout_file', 'submission_result_file'];
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts:145]: On the frontend, display them after capitalization and spacing. That is, column names (to be rendered): 'Status', 'Stderr File', 'Stdout File', 'Submission Result File'.

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.html [hunk 5]
```diff
@@ -69,102 +69,92 @@
       </div>
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmissions/challengesubmissions.component.html]: We already have a check mark, "Shown on leaderboard" in each collapsible. So is this needed now?

**Review by lunayach — CHANGES_REQUESTED** [review]
Functionality works! Minor changes needed.

File: src/app/app.module.ts (hunk 1 above)
Comment (sanketbansal) [reply src/app/app.module.ts]: Done

File: src/app/components/challenge/challenge.component.spec.ts (hunk 2 above)
Comment (sanketbansal) [reply src/app/components/challenge/challenge.component.spec.ts]: Done

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts (hunk 3 above)
Comment (sanketbansal) [reply src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts]: Done

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts (hunk 4 above)
Comment (sanketbansal) [reply src/app/components/challenge/challengesubmissions/challengesubmissions.component.ts:145]: Done

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.html (hunk 5 above)
Comment (sanketbansal) [reply src/app/components/challenge/challengesubmissions/challengesubmissions.component.html]: Removed that check once.

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.spec.ts [hunk 6]
```diff
@@ -11,6 +11,7 @@ import { HttpClientModule } from '@angular/common/http';
 import { ForceloginComponent } from '../../../components/utility/forcelogin/forcelogin.component';
 import { RouterTestingModule } from '@angular/router/testing';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
+import {MatTableModule} from '@angular/material';
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmissions/challengesubmissions.component.spec.ts]: Spacing.

Comment (sanketbansal) [reply src/app/components/challenge/challengesubmissions/challengesubmissions.component.spec.ts]: Done

File: src/app/components/challenge/challengesubmissions/challengesubmissions.component.html [hunk 7]
```diff
@@ -63,108 +63,94 @@
...
+                <div class="element">
+                  <span class="element-description-attribution">Submitted At</span>
+                  <span class="fw-regular element-description"> {{element.submitted_at | date:'medium'}} </span>
+                </div>
+                <div class="element">
+                  <mat-checkbox [checked]="element.is_public" [disabled]="element.status !== 'finished' || !selectedPhase['leaderboard_public']" (change)="changeSubmissionVisibility(element.id, element.is_public)">
+                    Show On Leaderboard
+                  </mat-checkbox>
+                  <a (click)="editSubmission(element)" class="element-description">
+                    <i class="fa fa-pencil"></i>
```
Comment (lunayach) [inline src/app/components/challenge/challengesubmissions/challengesubmissions.component.html:139]: Should we move the edit button to the next line after "Show On Leaderboard"?

Comment (sanketbansal) [reply src/app/components/challenge/challengesubmissions/challengesubmissions.component.html:139]: done

**Review by lunayach — APPROVED** [review]
@RishabhJain2018 LGTM!

**Pull request merged**